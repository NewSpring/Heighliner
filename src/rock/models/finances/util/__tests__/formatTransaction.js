import format from "../formatTransaction";

import {
  singleTransaction,
  singleACHTransaction,
  scheduleTransaction,
  multiFundScheduleTransaction,
  multipleTransactions,
} from "../__mocks__/sample-response";

jest.mock("moment", () => date => ({
  toISOString: () => `Mocked ISODate: ${date}`,
  subtract: (number, size) => `Mocked subtract ${number}, ${size}`,
}));

jest.mock("node-uuid", () => ({
  v4: jest.fn(() => "guid"),
}));

it("handles standard response data", () => {
  expect(
    format(
      {
        response: singleTransaction,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles multipleTransactions", () => {
  expect(
    format(
      {
        response: multipleTransactions,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles an authenticated response", () => {
  expect(
    format(
      {
        response: singleTransaction,
        person: { Id: 1, PrimaryAliasId: 2 },
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles adding a saved account", () => {
  expect(
    format(
      {
        response: singleTransaction,
        person: { Id: 1, PrimaryAliasId: 2 },
        accountName: "My Credit Card",
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("maps the source type value", () => {
  expect(
    format(
      {
        response: singleTransaction,
        origin: "http://example.com/give",
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("sets the schedule Id to be recovered", () => {
  expect(
    format(
      {
        response: singleTransaction,
        scheduleId: 10,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles an ACH transaction", () => {
  expect(
    format(
      {
        response: singleACHTransaction,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles a schedule", () => {
  expect(
    format(
      {
        response: scheduleTransaction,
        scheduleId: 10,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});

it("handles a schedule with multipleTransactions", () => {
  expect(
    format(
      {
        response: multiFundScheduleTransaction,
      },
      { Id: 3 },
    ),
  ).toMatchSnapshot();
});
