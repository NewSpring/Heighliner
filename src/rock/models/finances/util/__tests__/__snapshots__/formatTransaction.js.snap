exports[`test handles a schedule 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": 1111111111,
    "CurrencyTypeValueId": 157,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "ACH",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Id": 10,
    "IsActive": true,
    "NextPaymentDate": "Mocked ISODate: 20150101",
    "StartDate": "Mocked ISODate: 20150101",
    "TransactionFrequencyValue": Object {
      "amount": 1.02,
      "day-frequency": 1,
      "day-of-month": 1,
      "month-frequency": 2,
      "payments": 0,
      "plan-id": ";aljdflakf",
    },
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 128,
      "Amount": 1.02,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles a schedule with multipleTransactions 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": 1111111111,
    "CurrencyTypeValueId": 157,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "ACH",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
    "NextPaymentDate": "Mocked ISODate: 20150101",
    "StartDate": "Mocked ISODate: 20150101",
    "TransactionFrequencyValue": Object {
      "amount": 1.02,
      "day-frequency": 1,
      "day-of-month": 1,
      "month-frequency": 2,
      "payments": 0,
      "plan-id": ";aljdflakf",
    },
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 128,
      "Amount": 1.08,
      "Guid": "guid",
    },
    Object {
      "AccountId": 136,
      "Amount": 1000.54,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles adding a saved account 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Name": "My Credit Card",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "Id": 1,
    "PrimaryAliasId": 2,
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles an ACH transaction 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": 1111111111,
    "CurrencyTypeValueId": 157,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "ACH",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles an authenticated response 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "Id": 1,
    "PrimaryAliasId": 2,
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles multipleTransactions 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
    Object {
      "AccountId": 128,
      "AccountName": "Step Up Fund",
      "Amount": 3,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test handles standard response data 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test maps the source type value 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": "http://example.com/give",
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;

exports[`test sets the schedule Id to be recovered 1`] = `
Object {
  "Campus": Object {
    "Id": "3",
  },
  "FinancialPaymentDetail": Object {
    "AccountNumberMasked": "601160******6611",
    "CreditCardTypeValueId": 160,
    "CurrencyTypeValueId": 156,
    "Guid": "guid",
  },
  "FinancialPaymentValue": "Credit Card",
  "FinancialPersonSavedAccount": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "ReferenceNumber": "1573550949",
    "TransactionCode": "3388531907",
  },
  "Location": Object {
    "City": "Anderson",
    "PostalCode": "29621-3452",
    "State": "SC",
    "Street1": "808 Pebble Ln",
  },
  "Person": Object {
    "ConnectionStatusValueId": 67,
    "Email": "james.baxley@newspring.cc",
    "FirstName": "James",
    "Gender": 0,
    "Guid": "guid",
    "IsSystem": false,
    "LastName": "Baxley",
    "SystemNote": "Created from Apollos",
  },
  "Schedule": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Id": 10,
    "IsActive": true,
  },
  "SourceTypeValue": Object {
    "Url": undefined,
  },
  "Transaction": Object {
    "FinancialGatewayId": 3,
    "Guid": "guid",
    "Summary": "Reference Number: 3388531907",
    "TransactionCode": "3388531907",
    "TransactionDateTime": "Mocked subtract 5, hours",
    "TransactionTypeValueId": 53,
  },
  "TransactionDetails": Array [
    Object {
      "AccountId": 125,
      "AccountName": "General Fund",
      "Amount": 1,
      "Guid": "guid",
    },
  ],
}
`;
